test/vfd_swmr_common.c

```diff
@@ -227,58 +228,78 @@ await_signal(hid_t fid)
     }
 }
 
+/* Revised support routines that can be used for all VFD SWMR integration tests
+ */
+/* Initialize fields in config with the input parameters */
+void
+init_vfd_swmr_config(H5F_vfd_swmr_config_t *config, uint32_t tick_len, uint32_t max_lag, hbool_t writer,
+    hbool_t flush_raw_data, uint32_t md_pages_reserved, const char *md_file_fmtstr, ...)
+{
+    va_list ap;
+    
+    memset(config, 0, sizeof(H5F_vfd_swmr_config_t));
+
+    config->version = H5F__CURR_VFD_SWMR_CONFIG_VERSION;
+    config->pb_expansion_threshold = 0; 
+
+    config->tick_len = tick_len;
+    config->max_lag = max_lag;
+    config->writer = writer;
+    config->flush_raw_data = flush_raw_data;
+    config->md_pages_reserved = md_pages_reserved;
+
+    va_start(ap, md_file_fmtstr);
+    evsnprintf(config->md_file_path, sizeof(config->md_file_path),
```

Comment:
I'm trying to understand the code here.  Does `md_file_path` need memory allocation?

Reply:
Memory for H5F_vfd_swmr_config_t is allocated by the caller:

typedef struct H5F_vfd_swmr_config_t {
    int32_t     version;
    uint32_t    tick_len;
    uint32_t    max_lag;
    hbool_t     writer;
    hbool_t     flush_raw_data;
    uint32_t    md_pages_reserved;
    uint32_t    pb_expansion_threshold;
    char        md_file_path[H5F__MAX_VFD_SWMR_FILE_NAME_LEN + 1];
    char        log_file_path[H5F__MAX_VFD_SWMR_FILE_NAME_LEN + 1];
} H5F_vfd_swmr_config_t;

Reply:
I see.  Thanks.